feat: more http stuff, more tests

gkc opened the pull request:
**- What I did**
- feat: support query param `at_rt` in http GET requests to atServer. `…at_rt=meta` means get the metadata
- feat: root transmits query parameters to atServer in redirect URLs
- feat: support path-style or dot-style http requests - so GET /index.html.foo.bar.baz will work the same as GET /baz/bar/foo/index.html or GET /baz/bar/foo/html/index
- feat: infer the response content type based on the lookup key (previous example will set `text/html; charset=utf-8`). Default to `text/plain; charset=utf-8` for text data, `application/octet-stream` for binary data
- feat: support fetching of binary data. (Still TODO: functional tests for this feature)
- also support client specifying the content type via the `at_ct` (ct is for content-type) - `at_ct=application/json; charset=utf-8` for example
- test: lots of unit tests in atServer
- test: added redirect tests with query parameters to at_functional_tests
- build: updated tools/build_ve and tools/build_ee to export port 444 and set `httpsEnabled` to true by default in the atDirectory config

**- How I did it**
- See commits
- There are a few minor lint cleanups in there as well

**- How to verify it**
Tests pass

gkc committed:
feat: support query param `at_rt` in http GET requests to atServer. `at_rt=meta` means get the metadata
feat: root transmits query parameters to atServer in redirect URLs
feat: support path-style or dot-style http requests - so GET /index.html.foo.bar.baz will work the same as GET /baz/bar/foo/index.html or GET /baz/bar/foo/html/index

gkc committed:
test: fix test

gkc committed:
test: fix test

gkc committed:
test: fix test

gkc committed:
feat: interim: infer content-type of text/html if doing an http GET of a key which starts with 'html.' (i.e. you are getting something like foo/bar/index.html)
feat: interim: if supplied, set the response content-type to the value of the query parameter `at_ct=<content-type>`

cconstab reviewed (commented):
Probably a good idea to enable the 443 options in EE and VE

gkc committed:
build: update ve and ee builds to expose port 443 and set httpsEnabled to true in the atDirectory's (root server's) config

gkc committed:
feat: decode binary data from base64 to binary when we are returning a value which has `metadata.isBinary==true`
feat: infer content types when fetching data based on a pseudo path - so baz/bar/foo/index.html will be inferred as text/html. Using the `mime` dart package for this.

gkc committed:
build: at_root_server: update changelog and version in pubspec.yaml for atDirectory

gkc committed:
Merge remote-tracking branch 'origin/trunk' into more-http-atserver

gkc committed:
build: at_secondary_server: update changelog and version in pubspec.yaml for atDirectory

gkc commented:
> Probably a good idea to enable the 443 options in EE and VE

Done in [this commit](https://github.com/atsign-foundation/at_server/pull/2363/commits/22735727ce7127f8c84e6b957dc6f54d35245033)

gkc committed:
test: debugging tests

gkc committed:
test: debugging tests

gkc committed:
fix: at_root: handle emojis in redirect location header
feat: at_root: basic malformed request handling
feat: at_secondary: utf-8 charset for text mime types
feat: at_secondary: application/json mime type for metadata responses

gkc committed:
feat: switch atServer to log to stderr instead of stdout

gkc committed:
test: fixed bug in test

gkc committed:
test: fixed bug in test

gkc committed:
test: reverted temporary change in http_test.dart

cpswan committed:
Merge branch 'trunk' into more-http-atserver

gkc commented:
Setting back to draft until I've stopped the tests from flaking.

gkc committed:
fix: fix bug in getting binary data
test: retries while trying to figure out what's causing the flakiness in the `getAndCompareServerSigningKeyMetadata` functional test. Evidence so far is that it's always a problem with the Http response being empty despite there being a 200 statusCode

gkc committed:
Merge remote-tracking branch 'origin/more-http-atserver' into more-http-atserver

gkc committed:
test: fix test

gkc committed:
test: Add tests of http gets of binary data

gkc committed:
test: Fix functional test
